spec/unit/models/protocol_message_spec.rb

```diff
@@ -176,6 +176,14 @@ def new_protocol_message(options)
         end
       end
 
+      context 'when attach resumed flag' do
```

Comment:
Might be worth testing the negative case as well (if you create a `protocol_message` without the `attach_resume` flag then `has_attach_resume_flag?` returns `false`). A test like that would have made the accidental `&&` obvious.

Reply:
Fixed. Added two tests. One test with `flags = 0`:
`0 && 32 == 32` is `false`
`0 & 32 == 32` is `true`